src/lightning_app/utilities/proxies.py

```diff
@@ -372,20 +392,25 @@ def run_once(self):
 
         # 8. Deepcopy the work state and send the first `RUNNING` status delta to the flow.
         state = deepcopy(self.work.state)
+
+        # 9. Inform the flow the work is running and add the delta to the deepcopy.
         self.work._calls["latest_call_hash"] = call_hash
         self.work._calls[call_hash]["statuses"].append(make_status(WorkStageStatus.RUNNING))
-        self.delta_queue.put(ComponentDelta(id=self.work_name, delta=Delta(DeepDiff(state, self.work.state))))
+        delta = Delta(DeepDiff(state, self.work.state))
+        self.delta_queue.put(ComponentDelta(id=self.work_name, delta=delta))
 
-        # 9. Start the state observer thread. It will look for state changes and send them back to the Flow
-        # The observer has to be initialized here, after the set_state call above so that the thread can start with
-        # the proper initial state of the work
-        self.state_observer.start()
+        state += delta
 
         # 10. Unwrap the run method if wrapped.
         work_run = self.work.run
         if hasattr(work_run, "__wrapped__"):
             work_run = work_run.__wrapped__
 
+        # 11. Add the work state to the state observer and wait until it is ready.
+        self.state_observer._work_state.append(state)
+        while not self.state_observer._processing:
+            time.sleep(0.01)
```

Comment:
What if it is never ready? I'd add a timeout and generate an exception here.

Reply:
Sounds good.